gpu_ocean/SWESimulators/gpu_kernels/FBL_periodic_boundary.cu

```diff
@@ -38,133 +44,217 @@ __global__ void closedBoundaryUKernel(
     //Compute pointer to current row in the U array
     float* const U_row = (float*) ((char*) U_ptr_ + U_pitch_*tj);	
     
-    if ( (ti ==0 || ti == nx_) && tj < ny_halo_) {
+    if ( ( ((ti == 0  ) && (bc_west_ == 1)) || 
+           ((ti == nx_) && (bc_east_ == 1))    ) && tj < ny_ + 2) {
```

Comment:
Why  && tj < ny_ + 2  here)

Reply:
So that we don't write outside the buffer... why else? :)